xdsl/backend/riscv/lowering/convert_riscv_scf_to_riscv_cf.py

```diff
@@ -0,0 +1,166 @@
+from xdsl.dialects import builtin, riscv, riscv_cf, riscv_scf
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+class LowerRiscvScfForPattern(RewritePattern):
+    """
+    Create a CFG subgraph for the loop around its body blocks (if the body
+    contained other loops, they have been already lowered to a flow of blocks).
+    Maintain the invariants that a CFG subgraph created for any loop has a single
+    entry and a single exit, and that the entry/exit blocks are respectively
+    first/last blocks in the parent region.  The original loop operation is
+    replaced by the initialization operations that set up the initial value of
+    the loop induction variable (%iv) and computes the loop bounds that are loop-
+    invariant for affine loops.  The operations following the original scf.for
+    are split out into a separate continuation (exit) block. A condition block is
```

Comment:
[can you please follow standard loop terminology? you say `continuation` while in the diagram below and the code you label that as `end`. The "conventional" name for that block is `exit` (at least in LLVM) and I'd use that throughout (including generated labels)

More info here: https://llvm.org/docs/LoopTerminology.html

Reply:
This is exactly from [llvm](https://github.com/llvm/llvm-project/blob/aec73eade7af0e22c944714bec31570181bc1ad4/mlir/lib/Conversion/SCFToControlFlow/SCFToControlFlow.cpp#L50), but I'm happy to reword it to whatever you suggest :)

Reply:
Similarly for above, I didn't feel like rewriting this thing but it could definitely be made more approachable.

Reply:
I meant llvm "proper" not mlir; including a lot of compiler bibliography/projects and the LLVM doc I linked reflecting what llvm IR gets generated.
It might be approachable to a broader audience, that's my point.

I'll leave it up to you.